src/p2p/net_node.inl

```diff
@@ -1334,8 +1334,8 @@ namespace nodetool
       return true;
     if (get_incoming_connections_count() == 0)
     {
-      const el::Level level = el::Level::Warning;
-      MCLOG_RED(level, "global", "No incoming connections - check firewalls/routers allow port " << get_this_peer_port());
+      const el::Level level = el::Level::Info;
+      MCLOG_RED(level, "global", "No incoming connections - this does not affect the daemon's functionality. To enable incoming connections allow access from the Internet to port " << get_this_peer_port());
```

Comment:
It certainly affects it. It makes the connectivity more brittle.

Reply:
I've been using monerod like since the beginning of monero with NAT firewall in place, and I had absolutely no functionality problems. What kind of problems does this cause?

Reply:
It means the network connectivity is less dense: you can see that by imagining what would happen in the extreme case of this, if every monero node had broken incoming connections: then no node would be able to establish an outgoing connection either, and you wouldn't have a network at all. The more nodes with broken incoming connections, the more centralized the network on those nodes with working incoming connections.

Reply:
I am aware of that. It's just that I think the best way to approach this is to inform user honestly and ask them to make inbound connections available to contribute to the network, not scaring them. Because everyone has their reasons. 

(FYI I run a full node on a public vps, and my production machine does not.)

Also, I think [warning] log entries should log only problems that are critical.

Reply:
If it was critical, it's be ERROR or FATAL. Warnings are for things that aren't fatal but may need fixing. Info is for stuff that can be useful to know.
Anyway, as long as the warning is visible by default, I think it's fine. I'd rather it stays as warning, but if most people prefer it as info I'll be OK with that.

Reply:
I'm OK with the original message, but maybe it could be disabled if hide-my-port=1 and/or in-peers=0. That implies the user does not want incoming connections.

Reply:
Yeah, keeping the original message, but changing its log level to ` el::Level::Info` *if hide-my-port=1 and/or in-peers=0* sounds reasonable.

Reply:
It'a already off is offline or hide-my-port is set. Not if in-peers is 0 though.

Reply:
https://github.com/monero-project/monero/pull/4776